### PR TITLE
Allow `sklearn` versions > 1.3

### LINKED ISSUE #2216: Update `sklearn` Integration to Support Versions >1.3.0 and Resolve MLflow Autologging Issues
## Open Source Contributors Welcomed!
Please comment below if you would like to work on this issue!

### Contact Details [Optional]

support@zenml.io

### What happened?
The current ZenML Sklearn integration is restricted to versions of Sklearn <1.3.0, as defined in `src/zenml/integrations/sklearn/__init__.py`. However, the release of Sklearn 1.3.0 necessitates an update to this constraint. Additionally, this Sklearn version upgrade appears to cause issues with MLflow autologging, likely due to compatibility conflicts.

### Task Description
Update the Sklearn integration in ZenML to support Sklearn versions >1.3.0. Additionally, identify and resolve any issues arising in MLflow autologging due to this version update.

### Expected Outcome
- The Sklearn integration in ZenML should allow for the use of Sklearn versions >1.3.0.
- Any compatibility issues, especially with MLflow autologging, should be identified and resolved.
- Ensure that all tests, including CI pipelines, pass with the updated Sklearn version.

### Steps to Implement

- Modify the Sklearn version constraint in src/zenml/integrations/sklearn/__init__.py to allow for versions >1.3.0.
- Investigate and identify the root cause of the issues with MLflow autologging when using Sklearn 1.3.0.
- Implement necessary fixes or updates to ensure compatibility with the new Sklearn version.
- Thoroughly test the changes, especially focusing on MLflow autologging functionality.
- Update documentation and examples as necessary to reflect the support for the new Sklearn version.

### Additional Context
This update is crucial for keeping ZenML compatible with the latest machine learning tools and libraries, ensuring that users can leverage the newest features and improvements in Sklearn.

### Code of Conduct
- [ ] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vishal-Padia**
## Describe changes
I fixed #2216 to achieve that sklearn version > 1.3 is required to be installed.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

The changes in `tests/conftest.py` on lines 133 and 139 are just formatting changes that happen when I save the file in VSCode, I haven't changed anything inside it.
The change in `tests/conftest.py` is that I removed the deprecated `execfile` which was replaced with `File` in the new version.

## Summary by CodeRabbit

- **Documentation**
	- Updated the user guide to reflect the new command for installing the latest version of `sklearn`.

- **New Features**
	- Adjusted the `sklearn` integration to support versions above 1.3.

- **Tests**
	- Enhanced the testing configuration to align with updated dependencies and environment management.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: src/zenml/integrations/sklearn/__init__.py
Comment (strickvl) [inline src/zenml/integrations/sklearn/__init__.py:24]:
```suggestion
    REQUIREMENTS = ["scikit-learn"]
```

Since we're unblocking / releasing the requirement to versions > 1.3, let's not block the people who are on lower versions.

**Review by strickvl — CHANGES_REQUESTED** [review]
Thanks for the update. I think we don't want to restrict people to only use sklearn > 1.3. We just didn't want to prevent them any more, so we can probably just remove the restriction completely. Otherwise let's let the CI run through and we'll see.

File: tests/conftest.py
Comment (avishniakov) [inline tests/conftest.py:23]: Why this change is needed? Conftest is used across all test suites and we need to justify such a change somehow.

**Review by avishniakov — CHANGES_REQUESTED** [review]
Thanks for contributing! Agree with @strickvl comments and have one more question.

File: tests/conftest.py
Comment (strickvl) [reply tests/conftest.py:23]: In the PR description it mentions that `execfile` is deprecated, but agreed it's a sensitive change.

Comment (avishniakov) [reply tests/conftest.py:23]: I see, well let's give CI a shot then and see if something is affected

**Comment by strickvl** [comment]
@Vishal-Padia please run the formatting script to fix the linting failures.

**Comment by Vishal-Padia** [comment]
I'll run the formatting scripts to fix the linting failures.

**Comment by Vishal-Padia** [comment]
I am getting an error while running this `zenml integration install -y -i feast`
I'll just solve it and then run the formatting scripts

**Comment by strickvl** [comment]
Just run `pip install -e ".[server,dev]"` and that should be enough to run the formatting scripts.

**Comment by Vishal-Padia** [comment]
okay I ran that command and many libraries got installed.
These are the formatting scripts right:
```
bash scripts/format.sh
bash scripts/run-ci-checks.sh
bash scripts/test-coverage-xml.sh
```

**Comment by strickvl** [comment]
Just run this one: `bash scripts/format.sh` and then commit the changes and push

**Comment by Vishal-Padia** [comment]
okay understood

**Comment by Vishal-Padia** [comment]
It's done!
I pushed after running the formatting scripts.

File: tests/conftest.py
Comment (Vishal-Padia) [reply tests/conftest.py:23]: The `py._builtin` module was part of older versions of `pytest `and is now deprecated.

**Comment by strickvl** [comment]
> @strickvl I got a notification saying that the run got cancelled. Some of the tests failed

https://github.com/zenml-io/zenml/actions/runs/7503345737?pr=2271 here you can view the progress of the tests. It's ok so far!

**Comment by Vishal-Padia** [comment]
Hey, look all the checks passed

**Comment by strickvl** [comment]
Nice @Vishal-Padia! Just waiting for @avishniakov's approval and we'll get it merged!

**Review by avishniakov — APPROVED** [review]

**Pull request merged**
